Wrappers/Python/ccpi/optimisation/functions/FGP_TV.py

```diff
@@ -0,0 +1,275 @@
+        tmp_q = tmp_p.copy()
+        x = self.gradient.domain_geometry().allocate()     
+        p1 = tmp_p.copy()
+
+        for k in range(self.inner_iterations):
+                                                                                   
+            t0 = t
+            self.gradient.adjoint(tmp_q, out = x)
+            
+            # this can be replaced by axpby
+            x *= -self.regularising_parameter
+            x *= tau
+            x += b
+            self.projection_C(x, out = x)                       
+
+            self.gradient.direct(x, out=p1)
+            p1 *= 1./(8*self.regularising_parameter)
+            p1 /= tau
+            p1 += tmp_q
+            self.projection_P(p1, out=p1)
+            
+            if self.tolerance is not None:
+                
+                if k%5==0:
+                    error = (p1-tmp_q).norm()/p1.norm()            
+                    if error<=self.tolerance:                           
+                        break
+
+            t = (1 + numpy.sqrt(1 + 4 * t0 ** 2)) / 2                                                                                                                                             
+            tmp_q.fill(p1 + (t0 - 1) / t * (p1 - tmp_p))                                      
+            tmp_p.fill(p1)             
+                                        
+        if out is None:            
+            print("Run for {} iterations".format(k))
+            return self.projection_C(b - self.regularising_parameter*tau*self.gradient.adjoint(tmp_q))
+        else:
+            print("Run for {} iterations".format(k))            
+            self.gradient.adjoint(tmp_q, out = out)
+            out*=tau
+            out*=-self.regularising_parameter
+            out+=b
+            self.projection_C(out, out=out)
+    
+    def convex_conjugate(self,x):        
+        return 0.0    
+    
+if __name__ == '__main__':
+    
+    import numpy as np                         
+    import matplotlib.pyplot as plt
+
+#     from ccpi.optimisation.algorithms import PDHG
+
+#     from ccpi.optimisation.operators import BlockOperator, Identity, Gradient
+#     from ccpi.optimisation.functions import ZeroFunction, L1Norm, \
+#                           MixedL21Norm, BlockFunction, L2NormSquared,\
+#                               KullbackLeibler
+    from ccpi.framework import TestData
+    import os
+    import sys
+    from ccpi.plugins.regularisers import FGP_TV as CCPiReg_FGP_TV
+    from ccpi.filters import regularisers    
+    
+    loader = TestData(data_dir=os.path.join(sys.prefix, 'share','ccpi'))
+    data = loader.load(TestData.SHAPES)
+    ig = data.geometry
+    ag = ig
+
+    # Create noisy data. 
+    n1 = np.random.normal(0, 0.1, size = ig.shape)
+    noisy_data = ig.allocate()
+    noisy_data.fill(n1+data.as_array())
+
+    # Show Ground Truth and Noisy Data
+    plt.figure(figsize=(10,5))
```

Comment:
This line (plt.figure) should be repeated before each new image is plotted, for example before line 193. Currently, all the images are shown in the same figure, so only the last one is seen, along with lots of colorbars!